### PR TITLE
Add philosophy

### PR DESCRIPTION
**Pull request by antoineMoPa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,3 +60,20 @@ For a complete working project (with package.json, vite config, typescript confi
...
 ```
+
+# Philosophy
+
+We want to limit framefusion to 2 simple roles:
+
+ 1. Extracting frames from videos
+ 2. Combining frames into a video
+
+We want to design the API to be as minimal as possible to acheive these two goals.
```
Comment (michaeljherrmann) [inline README.md:71]:
```suggestion
We want to design the API to be as minimal as possible to achieve these two goals.
```

**Comment by michaeljherrmann** [comment]
> @michaeljherrmann mind having a look at this PR? This was before I setup the bot

remember you can still always use `/pr` even if it's not auto-posted 😅

**Review by michaeljherrmann — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (antoineMoPa) [reply README.md:71]: oops!

File: README.md [hunk 2]
```diff
@@ -60,3 +60,20 @@ For a complete working project (with package.json, vite config, typescript confi
...
+
+We also want to build the library in a way that we can provide different backends to acheive these goals, but still keep a similar interface.
+
+# Where we are with the philosophy currently
+
+Only point 1 is implemented so far (Extracting frames from videos).
+
+We also only provide a beamcoder backend and we still have to figure out how we can swap backends and skip compiling beamcoder. This could be through different npm packages.
```
[the remark this answers is not in the document]
Comment (dacox) [reply README.md:79]: @antoineMoPa isn't that a good thing, so we dont accidentally uninstall a version without  libav?

[the remark this answers is not in the document]
Comment (antoineMoPa) [reply README.md:79]: When using beamcoder, yes, but in the future, we might have different backends which don't require libav and beamcoder.

**Pull request merged**